siriuspy/siriuspy/orbintlk/main.py

```diff
@@ -1249,16 +1249,16 @@ def _do_callback_rxlock(self, pvname, value):
                         continue
                     # if not, it is a reliability failure
                     is_failure = True
+                    outs_in_failure.add(out)
                     self._update_log(f'ERR:{outnam} of {devname} not locked')
-            if not is_failure:
-                return
             # specifically for delta subsector (10), consider a failure only if
             # redundancy timing path is also not locked
             trigsrc = _LLTimeSearch.get_fout2trigsrc_mapping()[devname]
-            trigsrc_at_out = trigsrc[f'OUT{out}']
-            if trigsrc_at_out.sub[0:2] == '10':
-                rxlock = self._fout_dcct_dev['RxLockedLtc-Mon']
-                is_failure &= not _get_bit(rxlock, 0)  # out 0
+            for out in outs_in_failure:
+                trigsrcout = trigsrc[f'OUT{out}']
+                if trigsrcout.sub[0:2] == '10':
+                    rxlock = self._fout_dcct_dev['RxLockedLtc-Mon']
```

Comment:
I think it's a bit confusing to have it hardcoded here that the Fout with the redundancy is the one used for the DCCTs. Couldn't we add a member variable `self._fout_rintlk_dev = self._fout_dcct_dev` during initialization to make it clearer?

Reply:
deprecated

I moved this Fout to the dictionary of Fout devices and moved the redundancy pairs abstraction to a table, were we can populate with future pairs